rfc/Deposit fee to beneficiary.md

```diff
@@ -0,0 +1,65 @@
+- Another benefit is when fees left more than ED could be used to create the beneficiary account in case it does not exist.
+
+## Stakeholders
+
+## Explanation
+
+Currently the xcm only [deposit asset to the beneficiary](https://github.com/Snowfork/polkadot-sdk/blob/3f495e56ed01f24a29d341d8928c19cc2fd8f17e/bridges/snowbridge/primitives/router/src/inbound/mod.rs#L292-L293) and
+will change to ```DepositAsset { assets: Wild(All), beneficiary }``` which includes both the asset and fees.
+
+## Drawbacks
+
+This now places a burden on the destination parachain to store excess DOT in one of their pallets, if it is not configured to than the deposit will fail and assets will be trapped. It should be fine for AssetHub because relay token(i.e. DOT) is configured as a fungible asset as https://github.com/Snowfork/polkadot-sdk/blob/3f495e56ed01f24a29d341d8928c19cc2fd8f17e/cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs#L119 but may not be true for other parachains as usually no Parachain treats the system asset as the local currency.
+
+So we applies this change only on asset hub and not do this on the destination chain portion.
+
+## Testing, Security, and Privacy
+
+In https://github.com/Snowfork/snowbridge/pull/1174 we change the smoke test `send_token` to send weth to an non-existence account on asset hub, the test still works as expected.
+
+From the xcm log we can see that both the asset and the fee deposited to the beneficiary.
+
+```
+2024-04-16T02:50:18.942245Z TRACE xcm::barriers: AllowTopLevelPaidExecutionFrom origin: Location { parents: 1, interior: X1([Parachain(1013)]) }, instructions: [ReceiveTeleportedAsset(Assets([Asset { id: AssetId(Location { parents: 1, interior: Here }), fun: Fungible(10000000000) }])), BuyExecution { fees: Asset { id: AssetId(Location { parents: 1, interior: Here }), fun: Fungible(10000000000) }, weight_limit: Unlimited }, DescendOrigin(X1([PalletInstance(80)])), UniversalOrigin(GlobalConsensus(Ethereum { chain_id: 11155111 })), ReserveAssetDeposited(Assets([Asset { id: AssetId(Location { parents: 2, interior: X2([GlobalConsensus(Ethereum { chain_id: 11155111 }), AccountKey20 { network: None, key: [135, 209, 247, 253, 254, 231, 246, 81, 250, 188, 139, 252, 182, 224, 134, 194, 120, 183, 122, 125] }]) }), fun: Fungible(1000000000000000000) }])), ClearOrigin, DepositAsset { assets: Wild(All), beneficiary: Location { parents: 0, interior: X1([AccountId32 { network: None, id: [40, 72, 212, 220, 228, 167, 56, 124, 15, 56, 249, 50, 168, 47, 27, 78, 215, 131, 99, 62, 18, 255, 115, 240, 65, 108, 201, 130, 181, 216, 41, 48] }]) } }], max_weight: Weight { ref_time: 14541642000, proof_size: 362278 }, properties: Properties { weight_credit: Weight { ref_time: 0, proof_size: 0 }, message_id: Some([200, 234, 242, 47, 44, 176, 123, 172, 70, 121, 223, 10, 102, 14, 113, 21, 237, 135, 252, 253, 78, 50, 172, 38, 159, 101, 64, 38, 91, 187, 210, 111]) }
```

Comment:
Maybe just add the xcm program instead instead of all logs.
```
origin: Location { parents: 1, interior: X1([Parachain(1013)]) },
instructions: [
    ReceiveTeleportedAsset(
        Assets([Asset { id: AssetId(Location { parents: 1, interior: Here }), fun: Fungible(10000000000) }])
    ), 
    BuyExecution { fees: Asset { id: AssetId(Location { parents: 1, interior: Here }), fun: Fungible(10000000000) }, weight_limit: Unlimited },
    DescendOrigin(X1([PalletInstance(80)])),
    UniversalOrigin(GlobalConsensus(Ethereum { chain_id: 11155111 })),
    ReserveAssetDeposited(Assets([Asset { id: AssetId(Location { parents: 2, interior: X2([GlobalConsensus(Ethereum { chain_id: 11155111 }), AccountKey20 { network: None, key: [135, 209, 247, 253, 254, 231, 246, 81, 250, 188, 139, 252, 182, 224, 134, 194, 120, 183, 122, 125] }]) }), fun: Fungible(1000000000000000000) }])),
    ClearOrigin,
    DepositAsset { assets: Wild(All), beneficiary: Location { parents: 0, interior: X1([AccountId32 { network: None, id: [40, 72, 212, 220, 228, 167, 56, 124, 15, 56, 249, 50, 168, 47, 27, 78, 215, 131, 99, 62, 18, 255, 115, 240, 65, 108, 201, 130, 181, 216, 41, 48] }]) } }
],
max_weight: Weight { ref_time: 14541642000, proof_size: 362278 },
properties: Properties { 
  weight_credit: Weight { ref_time: 0, proof_size: 0 },
  message_id: Some([200, 234, 242, 47, 44, 176, 123, 172, 70, 121, 223, 10, 102, 14, 113, 21, 237, 135, 252, 253, 78, 50, 172, 38, 159, 101, 64, 38, 91, 187, 210, 111])
}

```

Reply:
https://github.com/Snowfork/snowbridge/pull/1174/commits/f7aa9614014d89109390b6dd85ab4e3e4093085f